shared-libs/tombstone-utils/src/tombstone-utils.js

```diff
@@ -0,0 +1,96 @@
+var _ = require('underscore');
+
+var TYPES = ['data_record', 'person', 'clinic', 'district_hospital', 'health_center'],
+    TOMBSTONE_TYPE = 'tombstone',
+    TOMBSTONE_ID_SEPARATOR = '____';
+
+module.exports = function(Promise, DB) {
+  var needsTombstone = function (doc) {
+    return TYPES.indexOf(doc.type) !== -1;
+  };
+
+  var generateTombstoneId = function (id, rev) {
+    return [id, rev, TOMBSTONE_TYPE].join(TOMBSTONE_ID_SEPARATOR);
+  };
+
+  var saveTombstone = function(doc, logger) {
+    var tombstone = _.omit(doc, ['_attachments', 'transitions', '_deleted']);
```

Comment:
I wouldn't bother removing "transitions" - it's being stored in a separate db as of 3.0 so won't be a big deal.